enterprise/server/util/redisutil/redisutil.go

```diff
@@ -68,43 +69,46 @@ func TargetToOptions(redisTarget string) *redis.Options {
 	}
 }
 
-func TargetToOpts(redisTarget string, useTLS bool) *Opts {
+func TargetToOpts(redisTarget string) *Opts {
 	if redisTarget == "" {
 		return nil
 	}
 	libOpts := TargetToOptions(redisTarget)
-	var tlsConfig *tls.Config
-	if useTLS {
-		tlsConfig = &tls.Config{
-			MinVersion: tls.VersionTLS12,
-		}
-	}
 	return &Opts{
 		Addrs:     []string{libOpts.Addr},
 		Network:   libOpts.Network,
 		Username:  libOpts.Username,
 		Password:  libOpts.Password,
 		DB:        libOpts.DB,
-		TLSConfig: tlsConfig,
+		TLSConfig: libOpts.TLSConfig,
 	}
 }
 
-func ShardsToOpts(shards []string, useTLS bool, username, password string) *Opts {
+func ShardsToOpts(shards []string, username, password string) *Opts {
 	if len(shards) == 0 {
 		return nil
 	}
-	var tlsConfig *tls.Config
-	if useTLS {
-		tlsConfig = &tls.Config{
-			MinVersion: tls.VersionTLS12,
+	var prevShardScheme string
+	for i, shard := range shards {
+		url, err := url.Parse(shard)
+		if err != nil {
+			log.Errorf("Failed to parse redis shard %q: %s", shard, err)
+			return nil
 		}
+
+		if i > 0 {
+			if url.Scheme != prevShardScheme {
+				log.Errorf("All redis shards must use the same url scheme, but found %q and %q", prevShardScheme, url.Scheme)
+				return nil
```

Comment:
returning `nil` seems like it will result in an NPE later on? should we just log and continue?

Reply:
Pushed a change to log + skip invalid url (failed `url.Parse`). But log fatal when the scheme is inconsistent between shards. This should be a sane "fail fast" default because the TLS connection gonna mess up regardless.